Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
+{
+    GameObject gameObject;
+    EventController.Action action; 
```

Comment:
Nice. Definitely on the right track with delegates. The one change I'd recommend is making this an event (this allows us to subscribe multiple actions at a time):

`
public event EventController.Action action;
void SubscribeActions () {
     action += firstAction;
     action += secondAction;
}
// This calls both firstAction and secondAction
void CallActions () {
    // You want this check to make sure there are currently delegates subscribed to this event
    if (action != null) {
        action();
    }
}
`